appcues/src/main/java/com/appcues/NavigationHandler.kt

```diff
@@ -0,0 +1,19 @@
+package com.appcues
+
+import android.net.Uri
+
+/**
+ * An handler that can be supplied by the host application to control behaviors around link navigation.
+ */
+public interface NavigationHandler {
+
+    /**
+     * Navigates to the given uri, suspending until navigation has completed. The navigation path
+     * value in `uri` may be a scheme link (deep link) inside of the current application, or an
+     * external web or application link.
+     *
+     * @param uri the uri for the destination to navigate.
+     * @return true if the navigation completed successfully, false if not.
+     */
+    suspend fun navigateTo(uri: Uri): Boolean
```

Comment:
Uri is probably the way to go but we could also go for String and have they setup any pattern they want for their screen navigation, but I'm ok using Uri too